lib/generate_letters.ml

```diff
@@ -1,42 +1,179 @@
+(* letter_deck is a list containing 7 random chars at any time. To be
+   used in a game of scrabble. If the char has been used in the game, a
+   new char will take its place, so there will never be an empty index.
+   Requires each char in the list to be an alphabetical letter at any
+   given time. *)
 type letter_deck = char list
-(** letter_deck is a list containing 7 random chars at any time. To be
-    used in a game of scrabble. If the char has been used in the game, a
-    new char will take its place, so there will never be an empty index.
-    Requires each char in the list to be an alphabetical letter at any
-    given time. *)
+
+(** [index_of_letter curr letter lst] is the index of the first
+    occurrence of this character in the list. Requires letter to be
+    present in lst and curr to be the current index (starting at 0).
+    index_of_letter : int -> char -> letter_deck -> int*)
+let rec index_of_letter curr letter = function
+  | [] -> raise (Failure "Empty List")
+  | h :: t ->
+      if h = letter then curr else index_of_letter (curr + 1) letter t
+
+(* vowel_amounts is an association list that links each vowel to the
+   probability of drawing that vowel given it's frequency in the tiles*)
+let vowel_amounts : (float * char list) list =
+  [
+    (12.1, [ 'U' ]);
+    (24.2, [ 'O' ]);
+    (27.2, [ 'A'; 'I' ]);
+    (36.3, [ 'E' ]);
+  ]
+
+(* consonant_amounts is an association list that links each consonant to
+   the probability of drawing that consonant given it's frequency in the
+   tiles*)
+let consonant_amounts : (float * char list) list =
+  [
+    (6.2, [ 'Q'; 'Z'; 'J'; 'X'; 'K' ]);
+    (12.5, [ 'F'; 'H'; 'V'; 'W'; 'Y'; 'B'; 'C'; 'M'; 'P' ]);
+    (18.7, [ 'G' ]);
+    (25., [ 'D'; 'S'; 'L' ]);
+    (37.5, [ 'T'; 'R'; 'N' ]);
+  ]
+
+(* alias table reference https://www.keithschwarz.com/darts-dice-coins/*)
+(* probability list needed for an alias table*)
+let prob = [ 0.42; 0.92; 0.56; 1.; 0.42; 0.06 ]
+
+(* alias list needed for an alias table*)
+let alias = [ 1; 2; 3; 0; 2; 2 ]
+
+(** [biased_bool prob total] is true for heads and false for tails. This
+    method simulates a biased coin, with a [prob]/[total] chance of
+    being heads. Requires [prob] <= [total].*)
+let biased_bool prob total =
```

Comment:
i'm confused why you need this function? like why is it biased?

Reply:
Alias tables use binary values with an unbalanced probability (ie; 0 is more likely to occur than 1). You can read about the methodology [here](https://www.keithschwarz.com/darts-dice-coins/) under "Generating Alias Tables".